File: Find duplicates in an array/sol.py
class Solution:
    def duplicates(self, arr, n): 
        al = []

        # Modification in the input array
        for i in range(n):
            arr[arr[i] % n] = arr[arr[i] % n] + n

        # Finding the duplicates
        for i in range(n):
            if arr[i] // n > 1:
                al.append(i)

        # Handling the case when no duplicates are found
        if not al:
            al.append(-1)

        return al
            

File: Find duplicates in an array/sol.cpp
class Solution{
  public:
    vector<int> duplicates(long long arr[], int n) {
        vector<int> al;
        for(int i=0; i<n; i++){
            arr[arr[i] % n] = arr[arr[i] % n] + n;
        }
        
        for(int i=0; i<n; i++){
            if((arr[i]/n > 1)){
                al.push_back(i);
            }
        }
        
        if(al.size() < 1){
            al.push_back(-1);
            return al;
        }
        
        return al;
    }
};
